File: ws-eclipse/projetos_udemy/curso_udemy/src/diagonal_negativos.java
import java.util.Locale;
import java.util.Scanner;

public class diagonal_negativos {
public static void diagonal_negativos() {
	
 Locale.setDefault(Locale.US);
 Scanner sc = new Scanner(System.in);
 
 int N, M, i , j, negativos;
 
 System.out.print("Quantas linhas vai ter a matriz? ");
 M = sc.nextInt();
 
 System.out.print("Quantas colunas vai ter a matriz? ");
 N = sc.nextInt();
 int[][] mat = new int[M][N];
 
 for (i = 0; i < M; i++) {
	 for (j = 0; j < N; j++) {
		 System.out.print("Elemento [" + i + "," + j + "]: ");
		 mat[i][j] = sc.nextInt();
 }
 } 
 
 System.out.println("\nMATRIZ DIGITADA:");
 for (i = 0; i < M; i++) {
	 for (j = 0; j < N; j++) {
		 System.out.println(mat[i][j] + " ");
	 }
	 System.out.println();
 }
 
 for (i = 0; i < M; i++) {
	 for (j = 0; j < N; j++) {
		 if(mat[i][j] < 0) {
			 negativos = negativos + 1;
		 }
 }
 }
 
 System.out.println("\nDIAGONAL PRINCIPAL:");
 System.out.println(mat[i][i]);
 
 System.out.println("QUANTIDADE DE NEGATIVOS : " + negativos);
 
 
 sc.close();

	}

}

File: ws-codeblocks/projetos_udemy_C++/diagonal_negativos.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{

int M, N, i, j, negativos, cont;

cout << ("Digite quantas linhas vai ter a matriz : ");
cin >> (M);
cout << ("Digite quantas colunas vai ter a matriz : ");
cin >> (N);

int mat[M][N];

for(i=0; i<M; i++){
    for(j=0; j<N; j++){
        cout << "Elemento : [ " << i << " , " << j << " ] : " ;
        cin >> mat[i][j];
    }
}

cout << "Diagonal principal : ";
for(int i = 0; i < N; i++){
   cout << mat[i][i] << " ";
}

for(i=0; i<M; i++){
    for(j=0; j<N; j++){
            if (mat[i][j] < 0){
                cont = cont + 1;
            }
    }
}

cout << "\n";
cout << "Quantidade de negativos : " << cont << "\n";

return 0;
}
